code/Peer.md

```diff
@@ -0,0 +1,173 @@
+# Peer
+
+Peer is the top-level entity in Storj. In the production network there's every peer runs in a standalone process. For tests multiple peers may run in a single process.
+
+Peers describe how a particular network participant is setup, wired together and run.
+
+Currently there network contains peers:
+
+1. Bootstrap
+2. Storage Node
+3. Satellite
+4. Uplink
+5. Version Control
+
+## Lifecycle
+
+In principle we can think of the full flow of Storage Node and Satellite as:
+
+1. Identity of the particular peer class is loaded. This is use to uniquely identify the peer on the network.
+2. A connection(s) to the database is made.
+3. Peer is created:
+3.1. Creation takes identity and database as an argument
+3.2. Listeners and servers are started.
+3.3. Every subsystem is created one by one:
+3.3.1. every sub-system consists of services and endpoints
+3.3.2. uses other services or databases as dependencies
+3.3.3. endpoints register themselves to the server
+4. Peer is started using `Run`, which:
+4.1. starts the services,
+4.2. starts waiting for network requests.
+5. Peer is running until,
+6. Peer is stopped via canceling the context used for `Run`.
+6.1. This cancels all subsystems in the reverse creation order.
+7. Peer is closed and all service and endpoint resources released.
+8. Database is closed
+
+Other peers may have a simplified setup.
+
+## Example Peer
+
+Here we show an facilitated Peer to show the general layout and how it works internally:
```

Comment:
"an" --> "a"

what do you mean by "facilitated" in this sentence?

Might be better as:
"Here is an example showing how a Peer works internally and its general layout.